File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/lock/setter/SetLockPositionUseCase.java
package com.lattis.ellipse.domain.interactor.lock.setter;

import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.lock.base.BaseLockUseCase;
import com.lattis.ellipse.domain.model.Lock;
import com.lattis.ellipse.domain.repository.BluetoothRepository;
import com.lattis.ellipse.domain.repository.LockRepository;

import javax.inject.Inject;

import io.reactivex.Observable;

/**
 * Created by ssd3 on 4/27/17.
 */

public class SetLockPositionUseCase extends BaseLockUseCase<Boolean> {

    private boolean locked;

    @Inject
    SetLockPositionUseCase(ThreadExecutor threadExecutor,
                           PostExecutionThread postExecutionThread,
                           BluetoothRepository bluetoothRepository,LockRepository lockRepository) {
        super(threadExecutor, postExecutionThread, bluetoothRepository,lockRepository);
    }

    public SetLockPositionUseCase withState(boolean locked) {
        this.locked = locked;
        return this;
    }

    public SetLockPositionUseCase forLock(Lock lock) {
        setScannedLock(lock);
        return this;
    }

    @Override
    protected Observable<Boolean> buildUseCaseObservable() {
        return setPosition(locked);
    }


}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/lock/setter/SetLockPositionUseCase.kt
package com.lattis.domain.usecase.lock.setter

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.BluetoothRepository
import com.lattis.domain.repository.LockRepository
import com.lattis.domain.usecase.lock.base.BaseLockUseCase
import com.lattis.domain.models.Lock
import com.lattis.domain.repository.SaSOrPSLockRepository
import com.lattis.domain.usecase.lock.connect.ConnectToLockUseCase
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class SetLockPositionUseCase @Inject internal constructor(
    threadExecutor: ThreadExecutor,
    postExecutionThread: PostExecutionThread,
    bluetoothRepository: BluetoothRepository, lockRepository: LockRepository,
    saSOrPSLockRepository: SaSOrPSLockRepository
) : BaseLockUseCase<Boolean>(
    threadExecutor,
    postExecutionThread,
    bluetoothRepository!!,
    lockRepository!!,
    saSOrPSLockRepository
) {
    private var locked = false
    private var fleetId:Int?=null

    fun withLockVendor(vendor:LockVendor): SetLockPositionUseCase {
        this.lockVendor = vendor
        return this
    }

    fun withState(locked: Boolean): SetLockPositionUseCase {
        this.locked = locked
        return this
    }

    fun forLock(lock: Lock): SetLockPositionUseCase {
        setScannedLock(lock)
        return this
    }

    fun withFleetId(fleetId: Int?): SetLockPositionUseCase {
        this.fleetId = fleetId
        return this
    }

    override fun buildUseCaseObservable(): Observable<Boolean> {
        return setPosition(lockVendor,locked,fleetId)
    }
}